p4_symbolic/symbolic/headers.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Contains helpers for creating, extracting, and managing concerete and
+// symbolic header structs.
+
+#include "p4_symbolic/symbolic/headers.h"
+
+#include "p4_symbolic/symbolic/util.h"
+
+namespace p4_symbolic {
+namespace symbolic {
+namespace headers {
+
```

Comment:
This is going to change: we will not have a FreeSymbolicHeader function anymore. SymbolicHeader will not be passed around internally. We will only use SymbolicMetadata, and at the end of the evaluation, we will extract SymbolicHeader from SymbolicMetadata to make it easier for client code to refer to these packet fields in assertions. Same with ConcereteHeader.

Reply:
already done on the other branch :D